Flames/flames_drv.py

```diff
@@ -293,78 +295,125 @@ def checkSequence(self, firingSequence):
         except Exception as e:
             print("Error: firingSequence is malformed or out of bounds", e)
             traceback.print_exc(file=sys.stdout)
-            return False 
+            return False
+
+    ## send bangCommandList to the poofer controller boards
+    def firePoofers(self, bangCommandList):
+        #TODO: This was just grabbed from heartbeat_controller. Need to
+        # make sure this is what we want
+        try:
+            if not self.isRunning:
+                return 1
+
+            if not ser:
+                ser = initSerial()
 
+            for command in bangCommandList:
+                ser.write(command.encode())
 
-    ## parse out firing sequence into bang code ##
-    def generateBangCommandList(self, firingSequence):
+        except Exception as e:
+            ser.close()
+            ser = None
+            print("Error sending bangCommandSequence to poofer controller boards", e)
+            traceback.print_exc(file=sys.stdout)
+
+    def disablePoofer(msgObj):
+        self.disabled_poofers.add(msgObj["name"])
+        event_manager.postEvent({"msgType":"poofer_disabled", "id":msgObj["name"]})
+
+    def enablePoofer(msgObj):
+        enabled_poofer = self.disabled_poofers.pop(msgObj["name"])
+        if enabled_poofer != None:
+            event_manager.postEvent({"msgType":"poofer_enabled", "id":msgObj["name"]})
+
+    def resumeAll():
+        self.isStopped = False
+        event_manager.postEvent({"msgType":"global_resume", "id":"all?"})
+
+    def stopAll():
+        self.isStopped = True
+        # TODO: send bang command to stop all poofers
```

Comment:
I'd consider setting static strings that just blast commands to all the controllers:
!0010~20~30~40~50~60~70~80.
...
!0310~20~30~40~50~60~70~80.

Remember that in the worst case, the poofers will shut off within 3 seconds, so even if you stop no poofers at all, things should be okay.

(In the worst worst case, there's a big red button that cuts power to the solenoids)